File: api/own-server-read.js
async function getOwnServerIp(key) {
    let targetUrl = "https://api.github.com/repos/ajg-machine" +
        "/connection-data/contents/own-server";
    let headers = { accept: "application/vnd.github.raw" };
    let response = await fetch(targetUrl, { headers });
    let data = await response.text();
    data = Uint8Array.from(atob(data), c => c.charCodeAt(0));
    key = await crypto.subtle.importKey(
        "raw", key, { name: "AES-GCM" }, false, ["encrypt", "decrypt"]
    );
    let algorithm = { name: "AES-GCM", length: 256, iv: data.slice(0, 12) };
    data = await crypto.subtle.decrypt(algorithm, key, data.slice(12));
    data = (new TextDecoder()).decode(data);
    return JSON.parse(data)["public-ip"];
}


async function main() {
    let key = atob(""); // ADD KEY
    key = Uint8Array.from(key, c => c.charCodeAt(0));
    console.log(await getOwnServerIp(key));
}


main();

File: api/own-server-read.py
def get_own_server_ip(key):
    from base64 import b64decode
    from cryptography.hazmat.primitives.ciphers.aead import AESGCM
    from json import loads
    from urllib.request import Request, urlopen
    target_url = \
        "https://api.github.com/repos/ajg-machine" + \
        "/connection-data/contents/own-server"
    headers = dict(accept="application/vnd.github.raw")
    request = Request(target_url, headers=headers)
    with urlopen(request) as response:
        data = b64decode(response.read().rstrip())
    data = AESGCM(key).decrypt(data[:12], data[12:], b"")
    return loads(data)["public-ip"]


from base64 import b64decode
key = b64decode(b"") # ADD KEY
print(get_own_server_ip(key))
